internal/featuretests/corspolicy_test.go

```diff
@@ -0,0 +1,341 @@
+		Resources: resources(t,
+			envoy.RouteConfiguration("ingress_http",
+				envoy.VirtualHost("hello.world",
+					&envoy_api_v2_route.CorsPolicy{
+						EnabledSpecifier: &envoy_api_v2_route.CorsPolicy_FilterEnabled{
+							FilterEnabled: &envoy_api_v2_core.RuntimeFractionalPercent{
+								DefaultValue: &envoy_type.FractionalPercent{
+									Numerator:   100,
+									Denominator: envoy_type.FractionalPercent_HUNDRED,
+								},
+							},
+						},
+						AllowOriginStringMatch: []*matcher.StringMatcher{{
+							MatchPattern: &matcher.StringMatcher_Exact{
+								Exact: "*",
+							}},
+						},
+						AllowCredentials: &wrappers.BoolValue{Value: true},
+						ExposeHeaders:    "custom-header-1,custom-header-2",
+					},
+					&envoy_api_v2_route.Route{
+						Match:  routePrefix("/"),
+						Action: routecluster("default/svc1/80/da39a3ee5e"),
+					}),
+			),
+		),
+		TypeUrl: routeType,
+	})
+
+	// Max Age
+	rh.OnAdd(fixture.NewProxy("simple").WithSpec(
+		projcontour.HTTPProxySpec{
+			VirtualHost: &projcontour.VirtualHost{
+				Fqdn: "hello.world",
+				CorsPolicy: &projcontour.CorsPolicy{
+					AllowOrigin:      []string{"*"},
+					AllowCredentials: true,
+					MaxAge:           "10m",
+				},
+			}, Routes: []projcontour.Route{{
+				Services: []projcontour.Service{{
+					Name: "svc1",
+					Port: 80,
+				}},
+			}},
+		}),
+	)
+
+	c.Request(routeType).Equals(&v2.DiscoveryResponse{
+		Resources: resources(t,
+			envoy.RouteConfiguration("ingress_http",
+				envoy.VirtualHost("hello.world",
+					&envoy_api_v2_route.CorsPolicy{
+						EnabledSpecifier: &envoy_api_v2_route.CorsPolicy_FilterEnabled{
+							FilterEnabled: &envoy_api_v2_core.RuntimeFractionalPercent{
+								DefaultValue: &envoy_type.FractionalPercent{
+									Numerator:   100,
+									Denominator: envoy_type.FractionalPercent_HUNDRED,
+								},
+							},
+						},
+						AllowOriginStringMatch: []*matcher.StringMatcher{{
+							MatchPattern: &matcher.StringMatcher_Exact{
+								Exact: "*",
+							}},
+						},
+						AllowCredentials: &wrappers.BoolValue{Value: true},
+						MaxAge:           "600",
+					},
+					&envoy_api_v2_route.Route{
+						Match:  routePrefix("/"),
+						Action: routecluster("default/svc1/80/da39a3ee5e"),
+					}),
+			),
+		),
+		TypeUrl: routeType,
+	})
+
+	// Virtual hosts with an invalid max age in their policy are not added
+	rh.OnAdd(fixture.NewProxy("simple").WithSpec(
+		projcontour.HTTPProxySpec{
+			VirtualHost: &projcontour.VirtualHost{
+				Fqdn: "hello.world",
+				CorsPolicy: &projcontour.CorsPolicy{
+					AllowOrigin:      []string{"*"},
+					AllowCredentials: true,
+					MaxAge:           "-10m",
+				},
+			}, Routes: []projcontour.Route{{
+				Services: []projcontour.Service{{
+					Name: "svc1",
+					Port: 80,
+				}},
+			}},
+		}),
+	)
+
+	c.Request(routeType).Equals(&v2.DiscoveryResponse{
+		Resources: resources(t,
+			envoy.RouteConfiguration("ingress_http")),
+		TypeUrl: routeType,
+	})
```

Comment:
You can check additional status here too:
```Go
c.Request(routeType).Equals(
    ...
).Status(obj).Like(...)
```

It's a bit inconvenient to have to pass an object here though.

Reply:
done!